neuralop/layers/transformer_block.py

```diff
@@ -0,0 +1,277 @@
+        self.mlp_expansion = mlp_expansion
+        self.non_linearity = non_linearity
+        self.norm = norm
+
+        self.query_basis = query_basis
+        self.query_siren_layers = query_siren_layers
+        self.query_fourier_scale = query_fourier_scale
+
+        self.lifting = nn.Linear(self.in_channels, self.hidden_channels) \
+            if self.in_channels != self.hidden_channels else nn.Identity()
+
+        self.out_norm = get_normalization(self.norm, self.hidden_channels)
+        self.to_out = MLPLinear([self.hidden_channels, self.hidden_channels, self.out_channels],
+                                non_linearity=self.non_linearity)
+
+        # build basis for decoder
+        if self.query_basis == 'siren':
+            self.query_basis_fn = SirenNet(dim_in=self.n_dim,
+                                           dim_hidden=self.hidden_channels,
+                                           dim_out=self.num_heads * self.head_n_channels,
+                                           num_layers=self.query_siren_layers)
+        elif self.query_basis == 'fourier':
+            self.query_basis_fn = nn.Sequential(
+                GaussianFourierFeatureTransform(self.n_dim,
+                                                mapping_size=self.head_n_channels,
+                                                scale=self.query_fourier_scale),
+                nn.Linear(self.head_n_channels * 2, self.num_heads * self.head_n_channels))
+        elif self.query_basis == 'linear':
+            self.query_basis_fn = nn.Linear(self.n_dim, num_heads * self.head_n_channels)
+        else:
+            raise ValueError(f'query_basis must be one of ["siren", "fourier", "linear"], got {self.query_basis}')
+
+        self.attention_layer = AttentionKernelIntegral(in_channels=self.hidden_channels,
+                                                        out_channels=self.hidden_channels,
+                                                        n_heads=self.num_heads,
+                                                        head_n_channels=self.head_n_channels,
+                                                        project_query=False)
+
+    def forward(self,
+                u,
+                pos_src,
+                pos_emb_module=None,
+                pos_qry=None,
+                **kwargs
+                ):
+        """
+           Project the input function u from the source grid to the query grid using the Transformer Decoder Block.
+
+          Parameters:
+                u: torch.Tensor, input tensor of shape [batch_size, num_src_grid_points, channels]
+                pos_src: torch.Tensor, grid point coordinates of shape [batch_size, num_src_grid_points, channels]
+                pos_emb_module: nn.Module, positional embedding module, by default None
+                pos_qry: torch.Tensor, grid point coordinates of shape [batch_size, num_sry_grid_points, channels],
+                         by default None and is set to pos_src
```

Comment:
It would be useful to include the specific purpose of `pos_qry` as compared to pos_src - inputs to the query basis function